### ISSUE TITLE
Default Keras callback for timing the training loop

### ISSUE THREAD

**Issue by Lokiiiiii** [keras-team/keras#16544]
If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.

TensorFlow version (you are using): nightly
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

The current Keras progress bar callback displays training latency in time/step. The current recommendation is to disable the progress bar in production workflows.

This poses 2 problems:

1. The metric time/step is not batch size agnostic. While tuning batch size to fit the H/W, metrics represented in terms of samples allow for better head to head comparison. Currently, an additional step of converting step to samples is required.

2. Production workflows that require continuous training will benefit greatly from being able to monitor training latency. A deterioration in training speed could lead to spike in training costs.

**Will this change the current api? How?**

No

**Who will benefit from this feature?**

Production workflows that require continuous training and need to monitor training speed.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): Yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

1. Reporting an additional metric for training latency based on samples. Eg: samples/sec

2. A new verbosity setting that allows for reporting the training latency metrics without the progress bar. This could even be part of the existing verbose=0.

**Comment by sushreebarsa** [comment]
@Lokiiiiii 
Could you please elaborate about your feature and please specify the use cases for this feature. Thank you!

**Comment by Lokiiiiii** [comment]
There is currently no way to choose ```samples``` from ```model.fit```.

Also I am interested in the metric without the Progress Bar, since the progress bar is never used in production environments - it adds a lot of junk log lines

**Comment by divyashreepathihalli** [comment]
you can choose `samples` in the progbarlogger and  add that callback to your `model.fit`. Is the information provided by  progbarlogger insufficient for your use case?

**Comment by rchao** [comment]
Hello @Lokiiiiii, here's how you can see progress by samples:

```
my_model.fit(...,
    verbose=1,
    callbacks=[tf.keras.callbacks.ProgbarLogger(count_mode='samples')])
```

To disable the progress bar, please use verbose=0.

Thanks for the question!

**Comment by SuryanarayanaY** [comment]
Hello, Thank you for reporting an issue.

We're currently in the process of migrating the new Keras 3 code base from keras-team/keras-core to keras-team/keras.
Consequently, This issue may not be relevant to the Keras 3 code base . After the migration is successfully completed, feel free to reopen this Issue at keras-team/keras if you believe it remains relevant to the Keras 3 code base. If instead this Issue is a bug or security issue in legacy tf.keras, you can instead report a new issue at [keras-team/tf-keras](https://github.com/keras-team/tf-keras/issues), which hosts the TensorFlow-only, legacy version of Keras.

To know more about Keras 3, please read https://keras.io/keras_core/announcement/
